File: core/src/main/java/net/torommo/logspy/ByLiteral.java
package net.torommo.logspy;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Configures a log spy to record log events to a logger for given name.
 */
@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface ByLiteral {
    /**
     * The name.
     *
     * @return The name.
     */
    String value();
}

File: core/src/main/kotlin/net/torommo/logspy/ByLiteral.kt
package net.torommo.logspy

/**
 * Configures a log spy to record log events to a logger for given name.
 * @param value the type
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class ByLiteral(val value: String)
